addOns/authhelper/src/main/java/org/zaproxy/addon/authhelper/internal/db/TableJdo.java

```diff
@@ -19,62 +19,144 @@
+        String user = getUser(db);
+        String password = getPassword(db);
+        ClassLoader classLoader = this.getClass().getClassLoader();
+        Flyway.configure(classLoader)
+                .table("AUTHHELPER_FLYWAY_SCHEMA_HISTORY")
+                .baselineOnMigrate(true)
+                .baselineVersion("0")
+                .dataSource(dbUrl, user, password)
+                .load()
+                .migrate();
 
-            Properties jdoProperties = new Properties();
-            jdoProperties.setProperty(Constants.PROPERTY_CONNECTION_URL, dbUrl);
-            jdoProperties.setProperty(Constants.PROPERTY_CONNECTION_USER_NAME, USER);
+        Properties jdoProperties = new Properties();
+        jdoProperties.setProperty(Constants.PROPERTY_CONNECTION_URL, dbUrl);
+        jdoProperties.setProperty(Constants.PROPERTY_CONNECTION_USER_NAME, user);
+        jdoProperties.setProperty(Constants.PROPERTY_CONNECTION_PASSWORD, password);
 
-            jdoProperties.put(PropertyNames.PROPERTY_CLASSLOADER_PRIMARY, classLoader);
+        jdoProperties.put(PropertyNames.PROPERTY_CLASSLOADER_PRIMARY, classLoader);
+
+        pmf = JDOHelper.getPersistenceManagerFactory(jdoProperties, "authhelper", classLoader);
+    }
 
-            pmf = JDOHelper.getPersistenceManagerFactory(jdoProperties, "authhelper", classLoader);
+    private static String getUrl(DatabaseServer db) throws DatabaseException {
+        try {
+            if (getUrlMethod != null) {
+                return (String) getUrlMethod.invoke(db);
+            }
+        } catch (Exception e) {
+            LOGGER.warn("An error occurred while getting the URL:", e);
+        }
 
+        try (Connection connection = getConnection(db)) {
+            return connection.getMetaData().getURL();
         } catch (SQLException e) {
             throw new DatabaseException(e);
         }
     }
 
+    private static Connection getConnection(DatabaseServer db) throws SQLException {
+        if (db instanceof ParosDatabaseServer pds) {
+            return pds.getNewConnection();
+        }
+        if (db instanceof ParosDatabaseServer pds) {
+            return pds.getNewConnection();
+        }
+        throw new SQLException("Unknown DB implementation");
+    }
+
+    private static String getUser(DatabaseServer db) {
+        try {
+            if (getUserMethod != null) {
+                return (String) getUserMethod.invoke(db);
+            }
+        } catch (Exception e) {
+            LOGGER.warn("An error occurred while getting the user:", e);
+        }
+
+        return "sa";
+    }
+
+    private static String getPassword(DatabaseServer db) {
+        try {
+            if (getPasswordMethod != null) {
+                return (String) getPasswordMethod.invoke(db);
+            }
+        } catch (Exception e) {
+            LOGGER.warn("An error occurred while getting the password:", e);
+        }
+
+        return "";
+    }
+
+    // @Override
```

Comment:
?

Will be needed when not using reflection?

Reply:
When compiling with the new core.